### ISSUE TITLE
getindex is not type stable for tables wider than 14 columns

### ISSUE THREAD

**Issue by AStupidBear** [JuliaComputing/IndexedTables.jl#176]
```
t = table([rand(rand([Int, Float32]), 10) for i in 1:14]...; names = [Symbol(Char('A' + i - 1)) for i in 1:14])
@code_warntype getindex(t, 1)
```
```
julia> @code_warntype getindex(t, 1)
Variables:
  #self# <optimized out>
  t::IndexedTables.NextTable{IndexedTables.Columns{NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N{Float32,Float32,Float32,Int64,Float32,Int64,Float32,Float32,Int64,Int64,Float32,Float32,Int64,Int64},NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N{Array{Float32,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Int64,1}}}}
  i::Int64

Body:
  begin
      $(Expr(:inbounds, false))
      # meta: location /BIGDATA1/highchain_ylu_1/shared/.julia/v0.6/IndexedTables/src/columns.jl getindex 232
      SSAValue(0) = (Core.getfield)((Core.getfield)(t::IndexedTables.NextTable{IndexedTables.Columns{NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N{Float32,Float32,Float32,Int64,Float32,Int64,Float32,Float32,Int64,Int64,Float32,Float32,Int64,Int64},NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N{Array{Float32,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Int64,1}}}}, :columns)::IndexedTables.Columns{NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N{Float32,Float32,Float32,Int64,Float32,Int64,Float32,Float32,Int64,Int64,Float32,Float32,Int64,Int64},NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N{Array{Float32,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Int64,1}}}, :columns)::NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N{Array{Float32,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Int64,1}}
      # meta: pop location
      $(Expr(:inbounds, :pop))
      return $(Expr(:invoke, MethodInstance for NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N{Float32,Float32,Float32,Int64,Float32,Int64,Float32,Float32,Int64,Int64,Float32,Float32,Int64,Int64}(::Float32, ::Float32, ::Float32, ::Int64, ::Float32, ::Int64, ::Float32, ::Float32, ::Int64, ::Int64, ::Float32, ::Float32, ::Int64, ::Int64), NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N{Float32,Float32,Float32,Int64,Float32,Int64,Float32,Float32,Int64,Int64,Float32,Float32,Int64,Int64}, :((Base.arrayref)((Core.getfield)(SSAValue(0), :A)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :B)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :C)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :D)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :E)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :F)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :G)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :H)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :I)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :J)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :K)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :L)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :M)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :N)::Array{Int64,1}, i)::Int64)))
  end::NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N{Float32,Float32,Float32,Int64,Float32,Int64,Float32,Float32,Int64,Int64,Float32,Float32,Int64,Int64}
```

However. when t has more than 14 columns, 
```
using JuliaDB
t = table([rand(rand([Int, Float32]), 10) for i in 1:15]...; names = [Symbol(Char('A' + i - 1)) for i in 1:15])
@code_warntype getindex(t, 1)
```
```
julia> @code_warntype getindex(t, 1)
Variables:
  #self# <optimized out>
  t::IndexedTables.NextTable{IndexedTables.Columns{NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N_O{Int64,Float32,Float32,Int64,Int64,Float32,Int64,Float32,Int64,Int64,Float32,Int64,Int64,Int64,Float32},NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N_O{Array{Int64,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Int64,1},Array{Float32,1}}}}
  i::Int64

Body:
  begin
      $(Expr(:inbounds, false))
      # meta: location /BIGDATA1/highchain_ylu_1/shared/.julia/v0.6/IndexedTables/src/columns.jl getindex 232
      SSAValue(0) = (Core.getfield)((Core.getfield)(t::IndexedTables.NextTable{IndexedTables.Columns{NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N_O{Int64,Float32,Float32,Int64,Int64,Float32,Int64,Float32,Int64,Int64,Float32,Int64,Int64,Int64,Float32},NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N_O{Array{Int64,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Int64,1},Array{Float32,1}}}}, :columns)::IndexedTables.Columns{NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N_O{Int64,Float32,Float32,Int64,Int64,Float32,Int64,Float32,Int64,Int64,Float32,Int64,Int64,Int64,Float32},NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N_O{Array{Int64,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Int64,1},Array{Float32,1}}}, :columns)::NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N_O{Array{Int64,1},Array{Float32,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Float32,1},Array{Int64,1},Array{Int64,1},Array{Int64,1},Array{Float32,1}}
      # meta: pop location
      $(Expr(:inbounds, :pop))
      return $(Expr(:invoke, MethodInstance for NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N_O{Int64,Float32,Float32,Int64,Int64,Float32,Int64,Float32,Int64,Int64,Float32,Int64,Int64,Int64,Float32}(::Int64, ::Float32, ::Float32, ::Int64, ::Int64, ::Float32, ::Int64, ::Float32, ::Int64, ::Int64, ::Float32, ::Int64, ::Int64, ::Int64, ::Float32), NamedTuples._NT_A_B_C_D_E_F_G_H_I_J_K_L_M_N_O{Int64,Float32,Float32,Int64,Int64,Float32,Int64,Float32,Int64,Int64,Float32,Int64,Int64,Int64,Float32}, :((Base.arrayref)((Core.getfield)(SSAValue(0), :A)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :B)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :C)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :D)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :E)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :F)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :G)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :H)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :I)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :J)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :K)::Array{Float32,1}, i)::Float32), :((Base.arrayref)((Core.getfield)(SSAValue(0), :L)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :M)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :N)::Array{Int64,1}, i)::Int64), :((Base.arrayref)((Core.getfield)(SSAValue(0), :O)::Array{Float32,1}, i)::Float32)))
  end::Any
```

**Comment by joshday** [comment]
Do you have a minimum reproducible example?

**Comment by shashi** [comment]
It's in the original post now.

**Comment by AStupidBear** [comment]
@shashi What do you mean?

**Comment by AStupidBear** [comment]
This is solved on Julia 0.7
